README.md

```diff
@@ -119,6 +120,8 @@ kubernetes {
     cadvisorSelector: 'job="kubernetes-cadvisor"',
     nodeExporterSelector: 'job="kubernetes-node-exporter"',
     kubeletSelector: 'job="kubernetes-kubelet"',
+    grafanaDashboardNamePrefix: 'Mixin / ',
+    grafanaDashboardTags: ['kubernetes', 'infrastucture'],
```

Comment:
This is specific to the configuration of these dashboards, what if a different mixin would have these fields? I think we should embed this in a config object specific to the kubernetes mixin.

Reply:
Just to double check my understanding - I thought the `config.libsonnet` in this repo was specific to the kubernetes-mixin. Are you saying different mixins could share the same `_config`?

Reply:
The design doc says all mixins use `_config` for configuration so if we didn't namespace the config for mixin specific things, it would be likely that other mixins would use the same field name, or append to it, while the "kubernetes" tag for example does not necessarily apply to any dashboards (and tags that apply to all dashboards doesn't sound like something that makes sense, as they may as well not be there then).

Reply:
I'm glad you called this potential conflict out. They feel already a bit namespaced to me, but we could add "k8s" to the two config vars `grafanaDashboard...` --> `grafanaK8sDashboard...` to avoid conflict with other mixins - does that seem better?

I'm not sure what you mean about the dashboard tags. Unless a dashboard explicitly sets it's tags and refers to the config var they won't be inheriting any tags from this change.

Reply:
That would work as well, although I can see more config knobs that only apply to this mixin, so I feel a config object dedicated to it would make sense either way.

Reply:
I like the nested object. I poked through the etcd, Consul, and vault mixins, looking for conflicts; didn't find any but it seems like the the grafana dashboards are the most likely to conflict. How about something like this? 
```
_config+:: {
    ...
    grafanaK8s :{
      // Grafana dashboard IDs are necessary for stable links for dashboards
      dashboardIDs: {
         ...
      },
      dashboardNamePrefix : "Kubernetes / ",
      dashboardTags: ["kubernetes-mixin"],

      // For links between grafana dashboards, you need to tell us if your grafana
      // servers under some non-root path.
      prefix: '',
    },
}
```
I started to wire it through the dashboards, not too bad if the above looks good

Reply:
I am still not clear on how the current config is not namespaced enough. To use the mixin you will have to import specifically the kubernetes mixin like `local kubernetes = import "kubernetes-mixin/mixin.libsonnet";`, so `_config` is then namespaced to the `kubernetes` variable. Maybe I am missing something, but I don't know why someone would overwrite the `_config` for another mixin with the `_config` from this mixin.

The suggested change above would break anyone who is overriding the `prefix`. Plus there are a lot more kubernetes-mixin specific variables at the root of `_config` already that people might be overriding. The project is alpha, so breaking changes are expected, but it is something to be aware of.

Reply:
The idea of mixins is that you just append each of them in your "downstream" for example:

```
local all = (import "kubernetes-mixin/mixin.libsonnet") +
  (import "my-internal/mixin.libsonnet);

// code to render everything
```

In that sense, I think the dashboardIDs should probably not be namespaced as any dashboard mixin should make use of it. `linkPrefix`, `dashboardNamePrefix`, `dashboardTags` are things that sound like they should be namespaced.

Reply:
Thanks for the example!

Reply:
Works for me - I namespaced `linkPrefix`, `dashboardNamePrefix`, and `dashboardTags` in a new commit.